src/redux/books/books.js

```diff
@@ -1,55 +1,65 @@
-import { v4 as uuidv4 } from 'uuid';
+import axios from 'axios';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 
-// Actions
 const ADDBOOK = 'bookstore/books/ADDBOOK';
 const REMOVEBOOK = 'bookstore/books/REMOVEBOOK';
+const GETBOOK = 'bookstore/books/GETBOOK';
 
-// Initial state
-const initialState = [
-  {
-    id: uuidv4(),
-    title: 'Gone with the Wind',
-    author: 'Margaret Mitchell',
-  },
-  {
-    id: uuidv4(),
-    title: 'Gifted Hands',
-    author: 'Ben Carson',
-  },
-  {
-    id: uuidv4(),
-    title: 'Rich Dad Poor Dad',
-    author: 'Robert Kiyosaki',
-  },
-];
-
-// Reducer
-const booksReducer = (state = initialState, action) => {
-  switch (action.type) {
-    case ADDBOOK:
-      return [...state, action.payload];
-    case REMOVEBOOK:
-      return state.filter((book) => book.id !== action.payload.id);
-    default:
-      return state;
+const apikey = '3NIoYi0HmR3w0D7JOHlP';
+const URL = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${apikey}/books`;
+
+export const getBooks = createAsyncThunk(GETBOOK, async () => {
+  const response = await axios.get(URL);
+  if (response.data) {
+    return response.data;
   }
-};
-
-// Action creators
-export const addBook = (book) => ({
-  type: ADDBOOK,
-  payload: {
-    id: uuidv4(),
-    title: book.title,
-    author: book.author,
-  },
+  return [];
+});
+
+export const addBook = createAsyncThunk(ADDBOOK, async (data) => {
+  await axios.post(URL, data);
+  return data;
+});
```

Comment:
- [ ] When I add books through the form in the browser, the book added is shown in the UI for as many books I add but when I refresh the page to check if the books added are stored on the API used, I see no books fetched or stored on the API to show in the browser. Could you work on the add book functionality to add and store a book on the API? 👍🏽